ci/test-appimagetool.sh

```diff
@@ -117,3 +117,21 @@ if [ "$hash1" != "$hash2" ]; then
     echo "Hash $hash1 doesn't match hash $hash2!"
     exit 1
 fi
+
+log "check --mksquashfs-opt forwarding"
+out=$("$appimagetool" appimagetool.AppDir appimagetool.AppImage --mksquashfs-opt "-misspelt-option" 2>&1)
+echo "${out}" | grep -q "invalid option"
+"$appimagetool" appimagetool.AppDir appimagetool.AppImage.1
+"$appimagetool" appimagetool.AppDir appimagetool.AppImage.2 --mksquashfs-opt "-mem" --mksquashfs-opt "100M"
+"$appimagetool" appimagetool.AppDir appimagetool.AppImage.3 --mksquashfs-opt "-all-time" --mksquashfs-opt "12345"
+hash1=$(sha256sum appimagetool.AppImage.1 | awk '{print $1}')
```

Comment:
Looks like escaping is missing here. Please run `shellcheck` on the script.

Reply:
Shellcheck is fine with this. Note that this line is copied verbatim from https://github.com/AppImage/AppImageKit/blob/672b3d0d1bf5fd74c3465da5fd62486606025a09/ci/test-appimagetool.sh#L114 :)
```
❯ shellcheck ci/test-appimagetool.sh

In ci/test-appimagetool.sh line 44:
REPO_ROOT="$(readlink -f "$(dirname "${BASH_SOURCE[0]}")"/..)"
^-------^ SC2034 (warning): REPO_ROOT appears unused. Verify use (or export if used externally).


In ci/test-appimagetool.sh line 45:
OLD_CWD="$(readlink -f .)"
^-----^ SC2034 (warning): OLD_CWD appears unused. Verify use (or export if used externally).


In ci/test-appimagetool.sh line 50:
export PATH="$(dirname "$appimagetool")":"$PATH"
       ^--^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In ci/test-appimagetool.sh line 60:
trap '[[ $FUNCNAME = exithook ]] || { last_lineno=$real_lineno; real_lineno=$LINENO; }' DEBUG
     ^-- SC2154 (warning): real_lineno is referenced but not assigned.

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- OLD_CWD appears unused. Verify us...
  https://www.shellcheck.net/wiki/SC2154 -- real_lineno is referenced but not...
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
```

I did actually try to move this into a small bash function, but that causes the trap script to report wrong line numbers on failure :/